### PR TITLE
[Iga] Map gauss results to embedded geometry process.

### PR DESCRIPTION
**Pull request by manuelmessmer**
**📝 Description**
This PR adds a mapping from gauss_results, which are calculated on a nurbs_volume (maine_model_part), to the nodes of the embedded geometry (embedded_model_part). Also a `vtk_embedded_geometry_output.h` ist added. The `VtkEmbeddedGeometryOutput` just calls `MapNurbsVolumeResultsToEmbeddedGeometryProcess` and then uses the standard `VtkOutputProcess`. New features are added to the unit test.

**🆕 Changelog**
Please summarize the changes in one list to generate the changelog:
- vtk_embedded_geometry_output
- map_nurbs_volume_results_to_embedded_geometry_process
- test_map_nurbs_volume_results_to_embedded_geometry_process

### CODE REVIEW

File: applications/IgaApplication/custom_processes/map_nurbs_volume_results_to_embedded_geometry_process.h [hunk 1]
```diff
@@ -43,13 +43,11 @@ class KRATOS_API(IGA_APPLICATION) MapNurbsVolumeResultsToEmbeddedGeometryProcess
     /// Pointer definition of MapNurbsVolumeResultsToEmbeddedGeometryProcess
     KRATOS_CLASS_POINTER_DEFINITION(MapNurbsVolumeResultsToEmbeddedGeometryProcess);
 
-    typedef Node                                             NodeType;
+    typedef Node                                                NodeType;
```
Comment (tteschemacher) [inline applications/IgaApplication/custom_processes/map_nurbs_volume_results_to_embedded_geometry_process.h:46]: That looks a bit weird on the phone. Would be alright though.

**Review by tteschemacher — APPROVED** [review]
I didn't see any obvious issues.

Let me know if you need a more comprehensive review.

File: applications/IgaApplication/custom_processes/map_nurbs_volume_results_to_embedded_geometry_process.h (hunk 1 above)
Comment (manuelmessmer) [reply applications/IgaApplication/custom_processes/map_nurbs_volume_results_to_embedded_geometry_process.h:46]: I just put NodeType to the correct indent.

File: applications/IgaApplication/custom_processes/map_nurbs_volume_results_to_embedded_geometry_process.cpp [hunk 2]
```diff
@@ -34,9 +34,38 @@ namespace Kratos
...
+        for (const std::string& r_variable_name : mThisParameters["nodal_results"].GetStringArray()){
+            if (KratosComponents<Variable<double>>::Has(r_variable_name)){
+                mDoubleVariableNode.push_back(&(KratosComponents< Variable<double>>::Get(r_variable_name)));
+            } else if (KratosComponents<Variable<array_1d<double, 3> >>::Has(r_variable_name)) {
+                mArrayVariableNode.push_back(&(KratosComponents<Variable<array_1d<double, 3>>>::Get(r_variable_name)));
+            } else {
+                KRATOS_ERROR << "Only double, and array variables are allowed in the variables list:"
+                    << " nodal_results.\n";
+            }
+        }
```
Comment (matekelemen) [inline applications/IgaApplication/custom_processes/map_nurbs_volume_results_to_embedded_geometry_process.cpp:48]: we desperately need an `std::variant`-based interface for `Variable`s

**Pull request merged**